src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -278,6 +278,28 @@ public void SetAllExceptions(enum_EXCEPTION_STATE dwState)
             }
         }
 
+        public int TryGetExceptionBreakpoint(string bkptno, out EXCEPTION_INFO exceptionInfo)
+        {
+            const string CppExceptionCategoryString = "{3A12D0B7-C26C-11D0-B442-00A0244A1DD2}";
+            exceptionInfo = new EXCEPTION_INFO();
+            ExceptionCategorySettings categorySettings;
+            if (_categoryMap.TryGetValue(new Guid(CppExceptionCategoryString), out categorySettings))
+            {
+                ulong breakpointNumber = Convert.ToUInt32(bkptno, CultureInfo.InvariantCulture);
+                if (categorySettings.CurrentRules.ContainsValue(breakpointNumber))
```

Comment:
>CurrentRules [](http://example.com/codeflow?start=37&length=12)

This violates the threading rule on line 97. We need to either start locking the dictionary.

Reply:
Okay. Will lock the dictionary.